src/DynamoCore/Engine/EngineController.cs

```diff
@@ -533,14 +533,14 @@ internal void ReconcileTraceDataAndNotify()
             var callsiteToOrphanMap = new Dictionary<Guid, List<string>>();
             foreach (var cs in liveRunnerServices.RuntimeCore.RuntimeData.CallsiteCache.Values)
             {
-                var orphanedSerializables = cs.GetOrphanedSerializables().ToList();
-                if (callsiteToOrphanMap.ContainsKey(cs.CallSiteID))
+                var orphanedSerializables = cs.GetOrphanedSerializables();
+                if (callsiteToOrphanMap.TryGetValue(cs.CallSiteID, out var serializablesForCallsite))
                 {
-                    callsiteToOrphanMap[cs.CallSiteID].AddRange(orphanedSerializables);
+                    serializablesForCallsite.AddRange(orphanedSerializables);
                 }
                 else
                 {
-                    callsiteToOrphanMap.Add(cs.CallSiteID, orphanedSerializables);
+                    callsiteToOrphanMap.Add(cs.CallSiteID, orphanedSerializables.ToList());
```

Comment:
so `orphanedSerializables` is already a `List<string>` - https://github.com/DynamoDS/Dynamo/blob/10df94ff5d0542328aee7205d6259aa0fb4f4921/src/Engine/ProtoCore/Lang/CallSite.cs#L475 though it's declared here as an `IList` - I assume that this calls the List constructor passing in the existing list, have you verified there is any benefit to moving this `ToList` call around?

If it's really such a big performance impact you could probably change this data structure to be a dict of `IList`